lib/create/create-issue-from-message.js

```diff
@@ -0,0 +1,175 @@
+const axios = require('axios');
+const IssueCreated = require('../messages/create/create-issue-attachment');
+const dialogs = require('../dialogs');
+const { Subscription } = require('../models');
+
+async function openDialog(req, res) {
+  const { slackWorkspace } = res.locals;
+  const {
+    trigger_id, team, channel, message,
+  } = req.body;
+
+  await slackWorkspace.client.dialog.open({
+    dialog: dialogs.createIssue(message, channel, team),
+    trigger_id,
+  });
+
+  return res.send();
+}
+
+async function getUserRepos(req, res) {
+  const { slackUser } = res.locals;
+  const { channel } = req.body;
+
+  const query = `
+      query {
+        viewer {
+          repositories(first: 30, orderBy: {field: UPDATED_AT, direction: DESC}) {
```

Comment:
We may get further here by using the `repositoriesContributedTo` query with the same `orderBy` and `includeUserRepositories`: true

https://developer.github.com/v4/object/user/#repositoriesContributedTo

We also need to find a better way to get the repositories in the channel, although that would require either an additional feature in the GitHub API or some additional caching, so leave that to me to figure out 😅

Reply:
True